src/ui-kit/form-controls/date/date.component.ts

```diff
@@ -315,18 +319,14 @@ export class SamDateComponent
     }
   }
 
-  getPossibleNum(value, event): number {
+  getPossibleNum(item, event): number {
     let possibleNum;
-    const ten = 10; // What is this number?
     const inputNum = KeyHelper.getNumberFromKey(event);
-    if (!isNaN(value)
-      && value !== '') {
-      possibleNum = (parseInt(value, undefined) * ten)
-        + inputNum;
-    } else {
-      possibleNum = inputNum;
+    if (this.keys.isAllowed(event)) {
+      const position = parseInt(item.selectionStart);
```

Comment:
@yerramshilpa Can you add a second arugment with `10` as the value to parseInt calls, e.g.,  `parseInt(item, 10)`? There are a couple of them I see in this PR

Reply:
Update